controller/helpers.js

```diff
@@ -0,0 +1,88 @@
+import fs from "fs";
+import config from "../config/config";
+
+const logDir = "controller/api/logs/"; //log events and errors on FIO side
+const serverErrLogsPathname = logDir + "Error.log"; //store the error startup and else unexpected errors error
+
+// function to handle all unexpected request errors (like bad internet connection or invalid response) and add them into Error.log file
+export const handleServerError = async (err, additionalMessage = null) => {
+    if (additionalMessage) console.log(additionalMessage+ ': ')
+    console.log(err.stack)
+
+    prepareLogDirectory(logDir, false);
+    await prepareLogFile({ filePath: serverErrLogsPathname }, false);
+
+    addLogMessage({
+        filePath: serverErrLogsPathname,
+        message: (additionalMessage ?  (additionalMessage + ': ') : '') + err.stack,
+    });
+}
+
+export const prepareLogDirectory = (directoryPath, withLogsInConsole = true) => {
+    if (fs.existsSync(directoryPath)) { //check if the log path exists
+        if (withLogsInConsole) console.log("The log directory exists.");
+    } else {
+        if (withLogsInConsole) console.log('The log directory does not exist.');
+        fs.mkdir(directoryPath, (err) => { //create new file
+            if(err) {
+                return console.log(err);
+            }
+            if (withLogsInConsole) console.log("The log directory was created!");
+        });
+    }
+}
+
+export const prepareLogFile = async ({
+    filePath,
+    blockName = null,
+    fetchLastBlockNumber = null,
+}, withLogsInConsole = true) => {
+    if (fs.existsSync(filePath)) { //check file exist
+        if (withLogsInConsole) console.log(`The file ${filePath} exists.`);
+        if (blockName) {
+            // todo: this could be improved to compare: lastBlockNumberInChain - lastProcessedBlockNumber <= InfuraApiPlanLimitation (or whatever else limitation value),
```

Comment:
The Infura API plan only allows a lookback of < 3500 blocks. So I recommend going back in steps of 3000.